components/microsoft_onedrive/actions/upload-file/upload-file.mjs

```diff
@@ -58,7 +58,7 @@ export default {
 
     const data = fs.readFileSync(filePath);
     const extension = (await fileTypeFromBuffer(data)).ext;
-    const name = !filename.includes(".")
+    const name = !filename.includes(".") && extension
       ? `${filename}.${extension}`
       : filename;
```

Comment:
It looks like the error (Cannot read properties of undefined (reading 'ext')) is actually coming from line 60:
`const extension = (await fileTypeFromBuffer(data)).ext;`

It's failing to get the `fileTypeFromBuffer`, so it considers looking for .ext as looking for the property of something that's undefined. As you noted, if the extension is provided in the filename, it shouldn't need to try to determine the extension. I would change the block of code in lines 59-63 to the following:

```
    const data = fs.reaeFileSync(filePath);
    let name = filename;
    if (!filename.includes(".")) {
      const extension = (await fileTypeFromBuffer(data)).ext;
      name = `${filename}.${extension}`;
    }
```